src/api/api.cpp

```diff
@@ -166,6 +167,24 @@ void set_output_names(tf_options& options, std::vector<const char*> names)
     options.output_node_names = std::vector<std::string>(names.begin(), names.end());
 }
 
+migraphx_status set_async_stream(execution_environment& e, void* stream)
+{
+    migraphx_status ret = migraphx_status_bad_param;
+    if(stream != nullptr)
+    {
+        e.queue = any_ptr(stream, "ihipStream_t");
```

Comment:
We shouldn't hardcode the typename, instead it should be passed from the C++ API level, we have `get_type_name<T>()` function to get the type name as a string:

```cpp
template<class Stream>
arguments run_async(const program_parameters& pparams, Stream s) const
{
    migraphx_arguments_t pout;
    call(&migraphx_program_run_async,
         &pout,
         this->get_handle_ptr(),
         pparams.get_handle_ptr(),
         s,
         get_type_name<Stream>().c_str());
    return arguments(pout, own{});
}
```

Reply:
fair, that would make sense. I got a fun surprise during initial testing when what I expected hipStream_t to be the same string wise and was surprised it was "ihipStream_t". I suppose that'll future proof against any other surprises if hip changes the typename?

Reply:
Done